packages/main/src/TableRowActionBase.ts

```diff
@@ -0,0 +1,115 @@
+import UI5Element from "@ui5/webcomponents-base/dist/UI5Element.js";
+import { customElement, property } from "@ui5/webcomponents-base/dist/decorators.js";
+import litRender from "@ui5/webcomponents-base/dist/renderer/LitRenderer.js";
+import TableRowActionBaseTemplate from "./generated/templates/TableRowActionBaseTemplate.lit.js";
+import TableRowActionBaseStyles from "./generated/themes/TableRowActionBase.css.js";
+import i18n from "@ui5/webcomponents-base/dist/decorators/i18n.js";
+import Icon from "./Icon.js";
+import Button from "./Button.js";
+import type Menu from "./Menu.js";
+import type MenuItem from "./MenuItem.js";
+import type Table from "./Table.js";
+import type TableRow from "./TableRow.js";
+import type TableRowAction from "./TableRowAction.js";
+import type I18nBundle from "@ui5/webcomponents-base/dist/i18nBundle.js";
+
+let MenuConstructor: new () => Menu;
+let MenuItemConstructor: new () => MenuItem;
+
+/**
+ * @class
+ * A class to serve as a foundation for table row actions.
+ * @constructor
+ * @abstract
+ * @extends UI5Element
+ * @since 2.7.0
+ * @public
+ */
+@customElement({
+	renderer: litRender,
+	styles: TableRowActionBaseStyles,
+	template: TableRowActionBaseTemplate,
+	dependencies: [Button, Icon],
+})
+
+abstract class TableRowActionBase extends UI5Element {
+	/**
+	 * Defines the visibility of the row action.
+	 *
+	 * **Note:** Hidden row actions still occupy space, allowing to hide the action while maintaining its position.
```

Comment:
Style / rephrasing proposal: 
```suggestion
	 * **Note:** Row actions can be hidden while their position is retained. However, hidden row actions still take up space.
```